script.js

```diff
@@ -1,4 +1,254 @@
+// global variables
+var winCount = 0;
+var playCount = 0;
+var gameMode = "Username Input";
+var userName = "";
+var modeChangeFlag = 0;
+var lastWinner = "";
+var lastKoreanWinner = "";
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
+  console.log(gameMode);
+  var myOutputValue = "";
+  if (gameMode == "Username Input") {
+    userName = input;
+    gameMode = "SPS mode";
+    return (
+      " Hello " +
+      userName +
+      "! Please type Scissors, Paper or Stone to play the game"
+    );
+  }
+
+  modeChangeChecker(input);
+  if (modeChangeFlag == 1) {
+    modeChangeFlag = 0;
+    return `Current mode is ${gameMode}`;
+  }
+
+  // plays game based on mode
+  if (gameMode == "SPS mode") {
+    myOutputValue = playGame(input.toLowerCase().trim(), SPS());
+  }
+  if (gameMode == "Reverse mode") {
+    myOutputValue = playReversedGame(input, SPS());
+  }
+  if (gameMode == "Computer mode") {
+    myOutputValue = playGame(SPS(), SPS());
+  }
+  if (gameMode == "Korean mode") {
+    if (lastKoreanWinner == "") {
+      myOutputValue = playKoreanGame(input, SPS());
+    } else if (lastKoreanWinner == userName) {
+      myOutputValue = playKoreanGame(input, SPS());
+      if (lastKoreanWinner == "Draw") {
+        winCount++;
+        playCount++;
+        myOutputValue +=
+          " You won the Korean game <br>" +
+          winRateMessage(winCount, playCount, userName);
+        lastKoreanWinner = "";
+      }
+    } else if (lastKoreanWinner == "Computer") {
+      myOutputValue = playKoreanGame(input, SPS());
+      if (lastKoreanWinner == "Draw") {
+        playCount++;
+        myOutputValue +=
+          " You lost the Korean game <br>" +
+          winRateMessage(winCount, playCount, userName);
+        lastKoreanWinner = "";
+      }
+    } else if (lastKoreanWinner == "Draw") {
+      myOutputValue = playKoreanGame(input, SPS());
+    }
+    console.log(lastKoreanWinner);
+  }
+
   return myOutputValue;
 };
+
+//function to generate random choice
+function SPS() {
+  const hands = ["scissors", "paper", "stone"];
```

Comment:
great job of using arrays here, which definitely makes the code for this cleaner as there is no need to use if else statements to give different hand output